lib/core/macros.nim

```diff
@@ -800,10 +800,19 @@ proc newLit*[T](s: set[T]): NimNode {.compileTime.} =
     var typ = getTypeInst(typeof(s))[1]
     result = newCall(typ,result)
 
-proc newLit*(arg: tuple): NimNode {.compileTime.} =
-  result = nnkPar.newTree
-  for a,b in arg.fieldPairs:
-    result.add nnkExprColonExpr.newTree(newIdentNode(a), newLit(b))
+proc isNamedTuple(T: typedesc): bool {.magic: "TypeTrait".}
+  ## See `typetraits.isNamedTuple`
+
+proc newLit*[T: tuple](arg: T): NimNode {.compileTime.} =
+  ## use -d:nimHasWorkaround14720 to restore behavior prior to PR, forcing
+  ## a named tuple even when `arg` is unnamed.
+  result = nnkTupleConstr.newTree
+  when defined(nimHasWorkaround14720) or isNamedTuple(T):
```

Comment:
Sometimes, I've thought that it would be nice if the compiler has a generic way to check for the presence of a bug fix (a bit like the generic `since` mechanism).

Right how, when I encounter an issue, I often leave a TODO comment in the code with a link to the filed issue. If each release of the compiler was storing internally an integer set of all the fixed issues, it would have been possible to write the following code:

```nim
when not hasFix(12231):
  # some work-around here:
else:
  {.fatal: "The work-around can be removed now".}
```

Reply:
@zah 
* this would not have helped here, where a bugfix causes some code relying on the bug to fail (see ggplotnim)
* nim already has a mechanism for that using `nimHasFoo` (see condsyms.nim), and it's strictly more flexible than an issue number, because it gives you freedom to specify what was fixed; eg if:
bug 1234 contains 2 sub issues `foo and `bar`, you can have a nim commit with: `nimHasFix1234Foo` (or `nimHasFixFoo`)

but we can use this convention when you need to branch in your code preemptively:

in issue 1234:
```nim
# test case in bug 1234
static: doAssert nimHasFix1234 # or some descriptive name, eg `nimHasFixFooBroken`
doAssert thisShouldWorkButFails()
```
and then a fix for issue 1234 shall add that test case in t1234.nim (or somewhere else) and define `nimHasFix1234`.

this allows you to preemptively branch on code that isn't fixed yet in the compiler/stdlib but expect it may in the near future, with similar benefits as `since`.

## stdlib vs compiler
* one subtlety is that the bugfix may occur either in the compiler or in stdlib:
  - if bug if fixed in compiler, `defineSymbol("nimHasFix1234")` must be in condsyms.nim
  - if bug if fixed in stdlib (as was the case here), `{.define(nimHasFix1234).}` must be in system.nim or maybe compilesettings.nim

The distinction matters, see why here: https://github.com/nim-lang/Nim/pull/14648

Reply:
there is such a library in rust: you can do it as a macro

Reply:
I can't find it :( but it checks the issue tracker on compile time e.g. in CI iirc. it's cool

Reply:
@alehander92 https://github.com/nikomatsakis/rfcs/blob/breaking-change-policy/text/0000-rustc-bug-fix-procedure.md ?

Reply:
no it was much simpler, just something like `waitIssue!(..)` 

https://www.reddit.com/r/rust/comments/gsq5gd/blocked_a_procmacro_that_emits_a_compiler_warning/

Reply:
sorry, it seems its `blocked!`